src/Common/Editor/Full_Site/Template_Utils.php

```diff
@@ -85,4 +90,123 @@ public static function inject_theme_attribute_in_content( $template_content ) {
 
 		return $template_content;
 	}
+
+	/**
+	 * Fetch a WP_Block_Template by the post_name and terms field.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $post_name The post_name field to search by.
+	 * @param string $terms     The terms field to search by, default to 'tec'.
+	 *
+	 * @return WP_Block_Template|null The newly created WP_Block_Template, or null on failure to locate.
+	 */
+	public static function find_block_template_by_post( string $post_name, string $terms = 'tec' ): ?WP_Block_Template {
+		// Let's see if we have a saved template?
+		$wp_query_args  = [
+			'post_name__in'  => [ $post_name ],
+			'post_type'      => 'wp_template',
+			'post_status'    => [ 'auto-draft', 'draft', 'publish', 'trash' ],
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+			'tax_query'      => [
+				[
+					'taxonomy' => 'wp_theme',
+					'field'    => 'name',
+					'terms'    => $terms,
+				],
+			],
+		];
+		$template_query = new WP_Query( $wp_query_args );
+		$posts          = $template_query->posts;
+
+		// If empty, our Block Template has not been created in `wp_posts` yet.
+		if ( empty( $posts ) ) {
+			return null;
+		}
+
+		$post = $posts[0] ?? null;
+
+		// Validate our query result.
+		if ( ! $post instanceof WP_Post ) {
+			do_action( 'tribe_log', 'error',
+				'Failed locating our Post for the Block Template', [
+					'method'    => __METHOD__,
+					'post_name' => $post_name,
+					'terms'     => $terms
+				] );
+
+			// Might as well bail, avoid errors below.
+			return null;
+		}
+
+		return self::hydrate_block_template_by_post( $post );
+	}
+
+	/**
+	 * Create a post for the wp_theme and return the hydrated WP_Block_Template.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $post_array Post array for insert.
+	 *
+	 * @return WP_Block_Template|null The newly created WP_Block_Template, or null on error.
+	 * @throws InvalidArgumentException
+	 */
+	public static function save_block_template( $post_array ): ?WP_Block_Template {
+		if ( empty( $post_array['post_name'] ) ) {
+			throw new InvalidArgumentException( "Must have both `post_name` parameter to denote this template uniquely." );
+		}
+		if ( empty( $post_array['tax_input'] ) ) {
+			throw new InvalidArgumentException( "Must have both `tax_input` parameter to include the term of the `wp_theme` this template is under." );
```

Comment:
"both" seems incorrect here
```suggestion
			throw new InvalidArgumentException( "Must have `tax_input` parameter to include the term of the `wp_theme` this template is under." );
```